Jenkinsfile

```diff
@@ -0,0 +1,142 @@
+@Library('socrata-pipeline-library')
+
+def deploy_environment = "staging"
+def service_sha = env.GIT_COMMIT
+
+// Stage control variables to simplify stage selection by build cause
+def stage_cut = false
+def stage_build = false
+def stage_dockerize = false
+def stage_deploy = false
+
+// Utility Libraries
+def sbtbuild = new com.socrata.SBTBuild(steps, "query-coordinator", "query-coordinator")
+def dockerize = new com.socrata.Dockerize(steps, "query-coordinator", BUILD_NUMBER)
+def deploy = new com.socrata.MarathonDeploy(steps)
+
+pipeline {
+  options {
+    ansiColor('xterm')
+  }
+  agent {
+    label 'build-worker'
+  }
+  environment {
+    PATH = "${WORKER_PATH}"
+  }
+
+  stages {
+    stage('Setup') {
+      steps {
+        script {
+          // checkout the repo
+          checkout scm
+
+          // determine what triggered the build and what stages need to be run
+          if (params.RELEASE_CUT == true) { // we're running a release cut
+            stage_cut = true  // other stages will be turned on in the cut stage if needed
+            deploy_environment = "rc"
+          }
+          else if (env.CHANGE_ID != null) { // we're running a PR test
+            stage_build = true
+          }
+          else if (BRANCH_NAME == "master") { // we're doing a master branch build for integration deploy
+            stage_build = true
+            stage_dockerize = true
+            stage_deploy = true
+          }
+          else {
+            // we're not sure what we're doing...
+            echo "Unknown build trigger - Exiting as Failure"
+            currentBuild.result = 'FAILURE'
+            return
+          }
+        }
+      }
+    }
+    stage('Cut') {
+      when { expression { return stage_cut == true } }
```

Comment:
Do you need the return here?

Reply:
I tried a bunch of different versions of a when statement in a test... and this is the only one that actually operated correctly off of a boolean.

In an ideal world it would just be `when { stage_cut }` but... that doesn't work in Jenkinsfile-groovy, sadly.

Reply:
Looks like there have been some upgrades... just re-ran the same test I'd done earlier, and now when { expression { <boolean> } } works.